docs/process-message/process-message.puml

```diff
@@ -0,0 +1,184 @@
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+else "Data does not respect the shape CommonMessageData"
+  InternalStorage -[#red]> ProcessMessage : Decode error
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+else "Storage returns data"
+  InternalStorage -[#green]> ProcessMessage : Data
+  deactivate InternalStorage
+end
+
+ProcessMessage -> profiles : Retrieve the latest version of recipient's profile
+activate profiles
+
+alt "Error from cosmos"
+  profiles -[#red]> ProcessMessage : 5xx from cosmos
+  ProcessMessage -[#red]>X CreatedMessage : Throw an error
+else "Empty response from cosmos"
+  profiles -> ProcessMessage : Empty response from cosmos
+  note left: User is not registered in IO
+  ProcessMessage -> messageStatus : Create a message-status REJECTED
+  activate messageStatus
+  note right:  The rejection_reason is \nUSER_NOT_FOUND
+  alt "Error from cosmos"
+    messageStatus -[#red]> ProcessMessage : 5xx from cosmos
+    ProcessMessage -[#red]>X CreatedMessage : Throw an error
+  end
+  ProcessMessage -> messageStatus : Set TTL to 3y for all versions
+  alt "Error from cosmos"
+    messageStatus -[#red]> ProcessMessage : 5xx from cosmos
+    deactivate messageStatus
+    ProcessMessage -[#red]>X CreatedMessage : Throw an error
+  end
+  ProcessMessage -> messages : Patch with TTL to 3y
+  activate messages
+  alt "Error from cosmos"
+    messages -[#red]> ProcessMessage : 5xx from cosmos
+    deactivate messages
+    ProcessMessage -[#red]>X CreatedMessage : Throw an error
+  end
+  ProcessMessage -> CreatedMessage: End
+else "Cosmos returns the profile"
+    profiles -[#green]> ProcessMessage : Profile
+    deactivate profiles
+end
+
+ProcessMessage -> ProcessMessage : Check if isInboxEnabled is true
+
+alt "isInboxEnabled is false"
+  ProcessMessage -> messageStatus : Create a message-status REJECTED
+  activate messageStatus
+  note right: The rejection reason is \nSERVICE_NOT_ALLOWED
+  alt "Error from cosmos"
+    messageStatus -[#red]> ProcessMessage : 5xx from cosmos
+    ProcessMessage -[#red]>X CreatedMessage : Throw an error
+  else "Cosmos updated the value"
+    messageStatus -[#green]> ProcessMessage : Ok response
+    deactivate messageStatus
+    ProcessMessage -> CreatedMessage : End
+  end
+end
+
+ProcessMessage -> servicePreferences : Ask for service preferences
+note right: We need service preferences\nto check if the recipient\nblocked some of the channels
+activate servicePreferences
+
+alt "Error from cosmos"
+  servicePreferences -[#red]> ProcessMessage : 5xx from cosmos
+  ProcessMessage -[#red]>X CreatedMessage : Throw an error
+else "Cosmos returns the service preferences"
+  servicePreferences -[#green]> ProcessMessage : Ok response
+  deactivate servicePreferences
+end
+
+ProcessMessage -> ProcessMessage : Calculate inbox and channels \nblocked from service preferences
+
+ProcessMessage -> ProcessMessage : Check if the sender service is SPECIAL
```

Comment:
Context needed for `SPECIAL`

Reply:
Take a look here https://github.com/pagopa/io-messages/pull/18/commits/2f2fcf0c565f27212f04f9b1e547f6703ed123be